js/ui/tree_view/ui.tree_view.base.js

```diff
@@ -1158,36 +1158,46 @@ const TreeViewBase = HierarchicalCollectionWidget.inherit({
             return true;
         }
 
-        if(!value && this._isLastRequired(node)) {
+        if(value === false && this._isLastRequired(node)) {
             if(this._showCheckboxes()) {
                 const $node = this._getNodeElement(node);
-                const checkbox = this._getCheckBoxInstance($node);
-
-                checkbox && checkbox.option('value', true);
+                this._setCheckboxValue($node, true);
```

Comment:
There are many checkboxes, let's rename this function to _setItemCheckboxValue

Reply:
I have removed the whole new method